syntax = "proto3";

package auth;

service Authentication {
  rpc Authenticate (AuthenticationRequest) returns (AuthenticationResponse) {}
  rpc Validate (ValidationRequest) returns (ValidationResponse) {}
  rpc Invalidate (InvalidationRequest) returns (InvalidationResponse) {}
  rpc Register (RegistrationRequest) returns (RegistrationResponse) {}
}

message AuthenticationRequest {
  string username = 1;
  string password = 2;
}

message AuthenticationResponse {
  string jws = 1;
}

message ValidationRequest {
  string jws = 1;
}

message ValidationResponse {
  string status = 1;
}

message InvalidationRequest {
  string jws = 1;
}

message InvalidationResponse {
  string status = 1;
}

message RegistrationRequest {
  string username = 1;
  string password = 2;
  string role = 3;
}

message RegistrationResponse {
  string status = 1;
  string uid = 2;
}


